### ISSUE TITLE
Multiple inference on a single image in reservoir.py

### ISSUE THREAD

**Issue by giorgionanfa** [Hananel-Hazan/bindsnet#195]
Hello, i would like to use the network present in reservoir.py to compute, for a single image, the output probabilities associated to each class. 
First of all, i perform the training of the network and then i save the weights and the state of the network with:

 torch.save(model.state_dict(), 'weights_res.pkl')
 network.save('Network.pt')

Hence, i comment the code relative to the training phase and i load the pre-trained network in this way:

network = load('Network.pt', learning=False)
model.load_state_dict(torch.load('weights_res.pkl')) 

The purpose of my analysis is to execute the inference on a single image of the MNIST dataset multiple times. I don't need to evaluate the accuracy of all the test images. So, when i perform the test phase, i modify the code in the following way:

loader = zip(poisson_loader(images, time=250), iter(labels))

n_iters = 500
test_pairs = []
for i, (datum, label) in enumerate(loader):
    **if i ==IMAGE_NUMBER:**                                                   #instead of if i % 100 == 0
        print('Test progress: (%d / %d)' % (i, n_iters))
    
    network.run(inpts={'I' : datum}, time=250)
test_pairs.append([spikes['O'].get('s').sum(-1), label])
...
...
for s, label in test_pairs:
    outputs = model(s)
    _, predicted = torch.max(outputs.data.unsqueeze(0), 1)
    **output_prob = nn.functional.softmax(torch.log(outputs-torch.min(outputs)+torch.max(outputs)))**

I use **outputs** in order to compute the output probabilities of the image IMAGE_NUMBER. My problem is that i obtain strange results: the output probabilities are often wrong and different every time i perform the test. For example, considering the image number 0, the first of the test set, i obtain the following results:

predicted tensor([5])
outputs tensor([-0.7257,  0.4272, -0.1043, -0.0577,  0.2966,  0.9772,  0.2493, -0.1010, -0.7273, -0.2445], grad_fn=<ThAddBackward>)
output probs log tensor([0.0575, 0.1251, 0.0939, 0.0967, 0.1175, 0.1574, 0.1147, 0.0941, 0.0574, 0.0857]

The label corresponding to the image number 0 is "7", so i don't understand why the probability relative to the class "7" is 0.0941, while the network recognizes the image (that is clearly a seven) as a five.
Maybe did i make a mistake during the test phase? Or is not this way correct to compute the output probabilities of an image? I hope i was clear to explain my aim and my problem.
Thank you in advance,
Giorgio

**Comment by djsaunde** [comment]
Could you send me your code? I could take a look at it.
